src/main/java/org/disq_bio/disq/impl/formats/bam/BamSink.java

```diff
@@ -107,18 +106,20 @@ public void save(
       out.write(BlockCompressedStreamConstants.EMPTY_GZIP_BLOCK);
     }
 
-    List<String> bamParts =
+    List<FileSystemWrapper.FileStatus> bamParts =
         fileSystemWrapper
-            .listDirectory(conf, tempPartsDirectory)
+            .listDirectoryStatus(conf, tempPartsDirectory)
             .stream()
-            .filter(new HiddenFileFilter())
+            .filter(fs -> new HiddenFileFilter().test(fs.getPath()))
+            .collect(Collectors.toList());
+    List<Long> partLengths =
+        bamParts
+            .stream()
+            .mapToLong(FileSystemWrapper.FileStatus::getLength)
+            .boxed()
             .collect(Collectors.toList());
-    List<Long> partLengths = new ArrayList<>();
-    for (String part : bamParts) {
-      partLengths.add(fileSystemWrapper.getFileLength(conf, part));
-    }
 
-    new Merger().mergeParts(conf, tempPartsDirectory, path);
+    new Merger(fileSystemWrapper).mergeParts(conf, bamParts, path);
```

Comment:
Does this make #43 more difficult to solve?  Adding another code path where `fileSystemWrapper` would need to be serialized, specifically.  Can `Merger` instantiate a new `fileSystemWrapper`?

Reply:
See #86

Reply:
I'm not sure what impact this has on #43/#86. It shouldn't make a difference should it?